### PR TITLE
include support for unquoted delimited lists during evaluation

### PR DESCRIPTION
**Pull request by shahms**
Also support `set(... CACHE)` and `get(... CACHE)` by just doing so locally.

These changes enable switching to using llvmbzlgen'd targets in Kythe (a PR which is forthcoming).

### CODE REVIEW

File: writer/starlark_test.go [hunk 1]
```diff
@@ -102,7 +103,7 @@ func TestReservedWord(t *testing.T) {
 	if err := writer.EndMacro(); err != nil {
 		t.Fatal("Unpexpected error ending macro: ", err)
 	}
-	expected := "def return_(ctx):\n    pass\n"
+	expected := "def return_(ctx):\n    return ctx\n"
```
Comment (creachadair) [inline writer/starlark_test.go]: `const expected = "…"`

**Review by creachadair — APPROVED** [review]

File: cmakelib/ast/eval.go [hunk 2]
```diff
@@ -125,3 +129,30 @@ func (v *VariableElement) Eval(vars Bindings) []string {
...
+			return "\r"
+		case 't':
+			return "\t"
+		default:
+			return m[1:]
+		}
+	})
+}
+
+// splitSemicolon splits the provided text on non-escaped semi-colons.
```
Comment (schroederc) [inline cmakelib/ast/eval.go]: Fix name

File: cmakelib/ast/eval.go [hunk 3]
```diff
@@ -125,3 +129,30 @@ func (v *VariableElement) Eval(vars Bindings) []string {
...
+func splitAndUnescape(text string) []string {
+	start := 0
```
Comment (schroederc) [inline cmakelib/ast/eval.go]: `var start int`

File: cmakelib/bindings/bindings.go [hunk 4]
```diff
@@ -74,8 +81,12 @@ func (m *Mapping) Get(key string) string {
 	return m.GetCache(key)
 }
 
-// GetCache returns the associated value from the variable cache (not implemented).
+// GetCache returns the associated value from the variable cache.
```
Comment (schroederc) [inline cmakelib/bindings/bindings.go]: or an empty string if not found

**Review by schroederc — APPROVED** [review]

**Pull request merged**